src/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static class PropertyVisibilityTests
+    {
+        [Fact]
+        public static void NoSetter()
+        {
+            var obj = new ClassWithNoSetter();
+
+            string json = JsonSerializer.ToString(obj);
+            Assert.Equal(@"{""MyString"":""DefaultValue""}", json);
+
+            ClassWithNoSetter objCopy = JsonSerializer.Parse<ClassWithNoSetter>(json);
+            Assert.Equal("DefaultValue", objCopy.MyString);
+        }
+
+        [Fact]
+        public static void NoGetter()
+        {
+            var objNoSetter = new ClassWithNoSetter();
+
+            string json = JsonSerializer.ToString(objNoSetter);
+            Assert.Equal(@"{""MyString"":""DefaultValue""}", json);
+
+            ClassWithNoGetter objNoGetter = JsonSerializer.Parse<ClassWithNoGetter>(json);
+            Assert.Equal("DefaultValue", objNoGetter.GetMyString());
+        }
+
+        [Fact]
+        public static void PrivateGetter()
+        {
+            var obj = new ClassWithPrivateSetterAndGetter();
+            obj.SetMyString("Hello");
+
+            string json = JsonSerializer.ToString(obj);
+            Assert.Equal(@"{}", json);
+        }
+
+        [Fact]
+        public static void PrivateSetter()
+        {
+            string json = @"{""MyString"":""Hello""}";
+
+            ClassWithPrivateSetterAndGetter objCopy = JsonSerializer.Parse<ClassWithPrivateSetterAndGetter>(json);
+            Assert.Null(objCopy.GetMyString());
+        }
+
+        // Todo: add tests with missing object property and missing collection property.
```

Comment:
Can you file an issue to improve test coverage here. Also to track your earlier comment: 
> Branch coverage is 80%-100% depending on file, but somewhat hampered because some tests are disabled because some of the APIs are not yet public (they were made internal). I do need more negative tests.